internal/pkg/runtime/engines/singularity/process_linux.go

```diff
@@ -386,7 +381,12 @@ func (engine *EngineOperations) PostStartProcess(pid int) error {
 		file.PPid = os.Getpid()
 		file.Image = engine.EngineConfig.GetImage()
 
-		namespaces := []struct {
+		// by default we add all namespaces except the user namespace which
+		// is added conditionally. This delegates checks to the C starter code
+		// which will determine if a namespace needs to be joined by
+		// comparing namespace inodes
+		path := fmt.Sprintf("/proc/%d/ns", pid)
+		for _, n := range []struct {
```

Comment:
I can't see why this change is needed (the fix in the comment is fine, thanks).

Is the variable name "namespaces" clashing with something else? Can you just rename the variable?

Reply:
I can restore it as it was before, I changed during tests